realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -22,6 +22,7 @@ using namespace tightdb;
 
 #if 1
 #define COL_TYPE_VALID(env,ptr,col, type)           TBL_AND_COL_INDEX_AND_TYPE_VALID(env,ptr,col, type)
+#define COL_TYPE_LINKLIKE(env,ptr,col)              TBL_AND_COL_INDEX_AND_LINKLIKE(env,ptr,col)
```

Comment:
Small nitpick, but wording like his makes me wonder what it means. LINK_OR_LINKLIST would be better imo. (although a bit long)